### PR TITLE
Configure keyboard matrix from info.json

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

This PR will generate `#define`'s based on information found in info.json. This is a big step towards making QMK easier to work with for non-programmers, and making it easier for collaborators to maintain a large number of keyboards using automated tooling. 

## Types of Changes

- [x] Core
- [x] New feature
- [x] Keyboard (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by skullydazed** [comment]
This is a cleaned up version of #10790. That one had too many problems due to commit history so I opened a new PR with a clean history.

**Comment by skullydazed** [comment]
For those who want to skip to the code you can look at commit 21983f4a9245c7ed0592228e037cf3ba5f91b843.

**Comment by skullydazed** [comment]
Given the proximity to develop closing I'm going to wait for the next cycle and beef this up a lot.

File: message.mk [hunk 1]
```diff
@@ -86,8 +86,7 @@ MSG_FILE_TOO_BIG = $(ERROR_COLOR)The firmware is too large!$(NO_COLOR) $(CURRENT
 MSG_FILE_TOO_SMALL = The firmware is too small! $(CURRENT_SIZE)/$(MAX_SIZE)\n
 MSG_FILE_JUST_RIGHT = The firmware size is fine - $(CURRENT_SIZE)/$(MAX_SIZE) ($(PERCENT_SIZE)%%, $(FREE_SIZE) bytes free)\n
 MSG_FILE_NEAR_LIMIT = The firmware size is approaching the maximum - $(CURRENT_SIZE)/$(MAX_SIZE) ($(PERCENT_SIZE)%%, $(FREE_SIZE) bytes free)\n
-MSG_PYTHON_MISSING = $(WARN_COLOR)WARNING:$(NO_COLOR)\n \
-	Can not run bin/qmk! This tool will be required when the develop branch is merged on 2020 Aug 29.\n\n\
+MSG_PYTHON_MISSING = $(ERROR_COLOR)ERROR:$(NO_COLOR) Can not run bin/qmk!\n\n\
 	Please run $(BOLD)util/qmk_install.sh$(NO_COLOR) to install all the dependencies QMK requires.\n\n
```
Comment (Erovia) [inline message.mk:90]:
```suggestion
	Please run $(BOLD)util/qmk_install.sh$(NO_COLOR) and $(BOLD)python3 -m pip install -U qmk$(NO_COLOR) to install all the dependencies QMK requires.\n\n
```

**Review by Erovia — APPROVED** [review]
Overall, looks good to me, especially for `develop`.

File: docs/reference_info_json.md [hunk 2]
```diff
@@ -49,25 +61,128 @@ All key positions and rotations are specified in relation to the top-left corner
...
-* `r`
-  * How many degrees clockwise to rotate the key.
-* `rx`
-  * The absolute position of the point to rotate the key around in the horizontal axis. Default: `x`
-* `ry`
-  * The absolute position of the point to rotate the key around in the vertical axis. Default: `y`
-* `ks`
-  * Key Shape: define a polygon by providing a list of points, in Key Units.
-  * **Important**: These are relative to the top-left of the key, not absolute.
-  * Example ISO Enter: `[ [0,0], [1.5,0], [1.5,2], [0.25,2], [0.25,1], [0,1], [0,0] ]`
```
[the remark this answers is not in the document]
Comment (skullydazed) [reply docs/reference_info_json.md:61]: They're not currently supported by CLI or web frontend.

File: message.mk (hunk 1 above)
Comment (skullydazed) [reply message.mk:90]: I believe we're already installing those as part of qmk_install. If not we should be...

**Pull request merged**